src/overlays/actors/ovl_En_Go2/z_en_go2.c

```diff
@@ -30,199 +216,2028 @@ const ActorInit En_Go2_InitVars = {
+                func_80034EC0(&this->skelAnime, sAnimations, 1);
+                this->actor.flags |= 1;
+                this->unk_26E = 2;
+                this->skelAnime.animPlaybackSpeed = 0.0f;
+                this->skelAnime.animCurrentFrame = this->skelAnime.animFrameCount;
+                EnGo2_GetItem(this, globalCtx, GI_CLAIM_CHECK);
+                this->actionFunc = EnGo2_InitGetItem;
+                this->goronState = 0;
+            }
+            break;
+    }
+}
+
+void EnGo2_GoronLinkStopRolling(EnGo2* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+
+    switch (this->goronState) {
+        case 0:
+            if (func_8010BDBC(&globalCtx->msgCtx)) {
+                return;
+            } else {
+                func_8010B680(globalCtx, 0x3031, NULL);
+                player->actor.freezeTimer = 10;
+                this->goronState++;
+            }
+        case 1:
+            break;
+        default:
+            return;
+    }
+
+    if (func_8010BDBC(&globalCtx->msgCtx) != 2) {
+        player->actor.freezeTimer = 10;
+    } else {
+        gSaveContext.infTable[16] |= 0x1000;
+        this->unk_26E = 1;
+        this->unk_211 = false;
+        this->isAwake = false;
+        this->actionFunc = EnGo2_CurledUp;
+    }
+}
+
+void EnGo2_GoronFireGenericAction(EnGo2* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+
+    Vec3s D_80A4854C = { 0x00, 0x00, 0x00 };
+
+    switch (this->goronState) {
+        // Wake up???
+        case 0:
+            if (func_8010BDBC(&globalCtx->msgCtx) == 2) {
+                EnGo2_GoronFireCamera(this, globalCtx);
+                globalCtx->msgCtx.msgMode = 0x37;
+                func_80034EC0(&this->skelAnime, sAnimations, 2);
+                this->waypoint = 1;
+                this->skelAnime.animPlaybackSpeed = 2.0f;
+                func_80A44D84(this);
+
+                this->actor.shape.rot = this->actor.posRot.rot;
+                this->animTimer = 60;
+                this->actor.gravity = 0.0f;
+                this->actor.speedXZ = 2.0f;
+                this->unk_194.unk_08 = D_80A4854C;
+                this->unk_194.unk_0E = D_80A4854C;
+                this->goronState++;
+                this->goronState++;
+                player->actor.posRot.rot.y = this->actor.posRot.rot.y;
+                player->actor.shape.rot.y = this->actor.posRot.rot.y;
+                player->actor.posRot.pos.x =
+                    (f32)((Math_Sins(this->actor.posRot.rot.y) * -30.0f) + this->actor.posRot.pos.x);
+                player->actor.posRot.pos.z =
+                    (f32)((Math_Coss(this->actor.posRot.rot.y) * -30.0f) + this->actor.posRot.pos.z);
+                func_8002DF54(globalCtx, &this->actor, 8);
+                func_800F5C64(0x51);
+            }
+            break;
+        // Walking away
+        case 2:
+            if (DECR(this->animTimer)) {
+                if (!(this->animTimer % 8)) {
```

Comment:
This isn't a boolean right? So it should be
```suggestion
                if ((this->animTimer % 8) == 0) {
```

Reply:
if ((this->animTimer % 8) == 0) doesn't match unfortunately, you need the negation `!`